bson/_cbsonmodule.c

```diff
@@ -2797,6 +2807,86 @@ static PyObject* _cbson_decode_all(PyObject* self, PyObject* args) {
     return result;
 }
 
+
+static PyObject* _cbson_array_of_documents_to_buffer(PyObject* self, PyObject* args) {
+    uint32_t size;
+    uint32_t value_length;
+    uint32_t position = 0;
+    int32_t initial_position;
+    buffer_t buffer;
+    const char* string;
+    PyObject* arr;
+    PyObject* result = NULL;
+    Py_buffer view = {0};
+
+    if (!PyArg_ParseTuple(args, "O", &arr)) {
+        return NULL;
+    }
+
+    buffer = pymongo_buffer_new();
+
+
+    if (!_get_buffer(arr, &view)) {
+        goto fail;
+    }
+
+    string = (char*)view.buf;
+    memcpy(&size, string, 4);
+    size = BSON_UINT32_FROM_LE(size);
+    /* save space for length */
+    if (pymongo_buffer_save_space(buffer, size) == -1) {
+        goto fail;
+    }
+    initial_position = pymongo_buffer_get_position(buffer);
```

Comment:
Can we remove initial_position? Isn't it always 0?

Reply:
No, it is not always 0

Reply:
The buffer was just created with pymongo_buffer_new() so when would it not be 0?

Reply:
I added a print statement and it was 4036

Reply:
Oh I get it, pymongo_buffer_save_space resizes the buffer and advances the buffer's position. We have to reset the position to 0 directly after the pymongo_buffer_save_space call.

Reply:
Done